File: RemoveElement.cpp
int removeElement(int* nums, int numsSize, int val)
{
  int i=0,j=numsSize-1,length=0;
  while( i<=j )
  {
    while (nums[i] != val && i<numsSize )
      ++i;
    while (nums[j] == val && j>=0 )
      --j;
    if( i-1 <= j )
    {
      int temp = nums[i];
      nums[i] = nums[j];
      nums[j] = temp;
      ++length;
    }
  }
  return length+1;
}
int main()
{
  int nums[2];
  nums[0]=nums[1]=3;
  int len = removeElement(nums,2,5);
  // nums[1]=nums[2]=2;
  // nums[0]=nums[3]=3;
  //int length = removeElement(nums,4,3);
  printf("%d\n",len);
  for (int i = 0; i < len; i++) {
    if(i==len-1)
      printf("%d\n",nums[i]);
    else
      printf("%d ",nums[i]);
  }
}

File: RemoveElement.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        index=0
        while( index < len(nums) ):
            if nums[index] == val:
                nums.pop(index)
            else:
                index=index+1
        return len(nums)
li=[3,3]
sol = Solution()
length = sol.removeElement(li,3)
print li
